src/System.Text.Json/tests/Serialization/DictionaryTests.cs

```diff
@@ -115,6 +115,13 @@ public static void DictionaryOfObject()
             Assert.Equal(@"{""Key1"":1}", json);
         }
 
+        [Fact]
+        public static void EmptyDictionaryOfObject()
+        {
+            Assert.Throws<JsonException>(() => JsonSerializer.Parse<Dictionary<string, string>>("\"headers\""));
+            Assert.Throws<JsonException>(() => JsonSerializer.Parse<string>("{}"));
```

Comment:
What is the exception message itself for these `JsonExceptions`? It would be good to validate they are meaningful.

Reply:
Both the exceptions are in this format
```c#
System.Text.Json.JsonException : The JSON value could not be converted to System.Collections.Generic.Dictionary`2[System.String,System.String]. Path: $ | LineNumber: 0 | BytePositionInLine: 9.
    Stack Trace:
      C:\git\corefx\src\System.Text.Json\src\System\Text\Json\ThrowHelper.Serialization.cs(84,0): at System.Text.Json.ThrowHelper.ThowJsonException(String message, Utf8JsonReader& reader, String path)
```